src/BSeries.jl

```diff
@@ -1453,4 +1456,419 @@ include("latexify.jl")
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                repeated_tree = true
+            end
+        end 
+        if repeated_tree == true
+            splice!(superarray, i)
+        end
+        repeated_tree = false
+    end
+
+    return superarray
+end
+
+function get_permutations(arr)
+    perms = collect(permutations(arr))
+    return perms
+end
+
+function num_leafs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
+    l = length(arrays)
+    for i in 1:l
+        arrays[i] = vcat([1], arrays[i])
+    end
+    return arrays
+end
+
+#this function multplies the coefficient for its symmetry factor
+function renormalize_bseries(coefficient_array,thetrees)
+    l = length(coefficient_array)
+    for i in 1:l
+        factor = 0
+        #because of the librarys we are using, some packages are repeated
+        #Then, we specify that the symmetry function comes from RootedTrees
+        factor = symmetry(thetrees[i])
+        coefficient_array[i] = coefficient_array[i]*(1//factor)
+    end
+    return coefficient_array
+end
+
+"""
+This function tells up to what order a method is Energy Preserving.
+"""
+function energy_preserving_order(A,b)
+    rka = RungeKuttaMethod(A, b)
+    p = 0
+    not_energy_preserving = false
+    while not_energy_preserving == false 
+#generate bseries 
+        if is_energy_preserving(rka,p+1) == false
+            not_energy_preserving = true
+        end
+        p = p + 1
+    end
+    return p-1
+end
+
+"""
+        energy_preserving_trees_test(trees,coefficients)
+
+this function is the applcation of  Theorem 2 of the paper 
+"Energy-Preserving Integrators and the Structure of B-series". 
+It takes an array 'trees' of level_sequences and the array 
+of coefficients. 
+Checks whether the sets of energy_preserving_tree satisfify the
+energy_preserving condition. 
+"""
+function energy_preserving_trees_test(trees, coefficients)
+    #for every tree, obtain the adjoint and check if it exists
+    length_coeff = length(trees)
+    #provided that a tree can have several adjoints, for every tree t we need to check if there is a linear combination of the coefficients 
+    #of its adjoints such that this linear combination is equal to the coefficient of t. 
+    #For this purpose, we cretae a MatrizEP to store all the information
+    MatrizEP = Vector{Int64}[]
+    error_searcher = false
+    for t in trees
+        if !isempty(t) 
+            #save the index of the level_sequence: this will be used for creating the matrix EP
+            highindex = indexator(trees,t)
+            #check if the level_sequence corresponds to a bush
+            bushflag = bush_detector(t)
+            #if it does, then the coefficient must be equal to zero.
+            if bushflag == true
+                if coefficients[highindex] != 0
+                    error_searcher = true
+                end
+            else
```

Comment:
This could just be
```
            # Check whether the level sequence corresponds to a bushy tree.
            # If so, we can exit early since the coefficients of bushy trees must be
            # zero in the modified equation of energy-preserving methods.
            if bush_detector(t)
                return false
            end
```

Reply:
I modified the comments and the structure of this function.